quic/s2n-quic-core/src/number.rs

```diff
@@ -257,3 +257,62 @@ upcast_usize!("16", u16, u16);
 upcast_usize!("32", u32, i32);
 upcast_usize!("64", u64, i64);
 upcast_usize!("128", u128, i128);
+
+/// A rational number represented by a numerator and a denominator
+pub struct Fraction(u32, u32);
```

Comment:
Is there a reason not to use the [`num_rational`](https://docs.rs/num-rational/latest/num_rational/struct.Ratio.html) crate for this? The nice thing is reduces ratios automatically so you don't end up overflowing too soon.

Reply:
Ah cool, I'll try it out